planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp

```diff
@@ -190,13 +194,16 @@ class SceneModuleInterface
 
   MarkerArray getDebugMarker() { return debug_marker_; }
 
+  AvoidanceInfoArray::SharedPtr getAvoidanceInfoArray() { return avoidance_info_array_ptr_; }
+
 private:
   std::string name_;
   rclcpp::Logger logger_;
 
 protected:
   MarkerArray debug_marker_;
   rclcpp::Clock::SharedPtr clock_;
+  mutable AvoidanceInfoArray::SharedPtr avoidance_info_array_ptr_{};
```

Comment:
@zulfaqar-azmi-t4 And sorry, one more comment. If you can use `mutable`, please let the variable name be `debug_*` so that we can see it is not related to the actual behavior and omit the time to investigate this variable flow.

Reply:
Thank you. I've changed the naming to `debug_avoidance_info_array_ptr_` in the new commit.